test/jdk/javax/swing/border/EtchedBorder/ScaledEtchedBorderTest.java

```diff
@@ -0,0 +1,240 @@
+        int expected = (int) Math.floor(scaling);
+        if (thickness != expected) {
+            throw new RuntimeException("Unexpected " + orientation + " Border thickness at x:"
+                                       + x + " y: " + y + ". Expected: " + expected + " Actual: " + thickness);
+        }
+    }
+
+    private static void checkVerticalBorder(int x, BufferedImage img, double scaling) {
+        int thickness = 0;
+        boolean checkShadow = false;
+        boolean checkHighlight = false;
+        for (int y = 0; y < img.getHeight(); y++) {
+            int color = img.getRGB(x, y);
+            if (!checkHighlight && !checkShadow) {
+                if (color == shadow.getRGB()) {
+                    checkHighlight = true;
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    throw new RuntimeException("Vertical Border was clipped or overdrawn.");
+                }
+            } else if (checkHighlight) {
+                if (color == shadow.getRGB()) {
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkHighlight = false;
+                    checkShadow = true;
+                    thickness = 1;
+                } else {
+                    throw new RuntimeException("Vertical Border has empty space between highlight and shadow.");
+                }
+            } else {
+                if (color == shadow.getRGB()) {
+                    throw new RuntimeException("Border colors reversed.");
+                } else if (color == highlight.getRGB()) {
+                    thickness++;
+                } else {
+                    verifyThickness(x, y, thickness, scaling, "Vertical");
+                    checkShadow = false;
+                    thickness = 0;
+                }
+            }
+        }
+    }
+
+    private static void createGUI(boolean show) {
+        // Render content panel
+        contentPanel = new JPanel();
+        contentPanel.setLayout(new BoxLayout(contentPanel, BoxLayout.Y_AXIS));
+
+        Dimension childSize = null;
+        for (int i = 0; i < 4; i++) {
+            childPanel = new JPanel(new BorderLayout());
+            childPanel.setBorder(BorderFactory.createCompoundBorder(
+                    BorderFactory.createEmptyBorder(0, i, 4, 4),
+                    BorderFactory.createEtchedBorder(highlight, shadow)));
+            childPanel.add(Box.createRigidArea(SIZE), BorderLayout.CENTER);
+
+            contentPanel.add(childPanel);
+            if (childSize == null) {
+                childSize = childPanel.getPreferredSize();
+                childPanel.setBounds(0, childSize.height * i,
+                        childSize.width, childSize.height);
+            }
+
+            contentPanel.setSize(childSize.width, childSize.height * 4);
```

Comment:
`contentPanel.setSize` is to be called after the for-loop when all child panels are added.